pcs/module_manager.py

```diff
@@ -128,57 +63,83 @@ def get_object(self):
         setattr(managed_obj, "__component__", self)
         return managed_obj
 
-    def _build_virtual_env(self) -> VirtualEnv:
-        # Only the root Module will setup and activate the VirtualEnv
-        if not self._use_venv or self._parent_modules:
-            return NoOpVirtualEnv()
-        req_paths = set()
-        for c in self.dependency_list(
-            include_parents=True, filter_by_types=[Module]
-        ):
-            if c.requirements_path is None:
-                continue
-            for req_path in str(c.requirements_path).split(";"):
-                full_req_path = self.repo.get_local_file_path(
-                    req_path, c.version
-                ).absolute()
-                req_paths.add(full_req_path)
-        return VirtualEnv.from_requirements_paths(req_paths)
-
-    def _handle_repo_spec(self, repos):
-        existing_repo = repos.get(self.repo.name)
-        if existing_repo:
-            if self.repo.to_dict() != existing_repo:
-                self.repo.name = str(uuid.uuid4())
-        repos[self.repo.name] = self.repo.to_dict()
-
-    def to_versioned_module(self, force: bool = False) -> "Module":
-        repo_url, version = self.repo.get_version_from_git(
-            self.file_path, version=self.version, force=force
-        )
-        prefixed_file_path = self.repo.get_prefixed_path_from_repo_root(
-            version, self.file_path
-        )
-        prefixed_reqs_path = None
-        if self.requirements_path:
-            prefixed_reqs_path = self.repo.get_prefixed_path_from_repo_root(
-                version, self.requirements_path
-            )
-        clone = Module(
-            repo=GitHubRepo(url=repo_url),
-            file_path=prefixed_file_path,
-            version=version,
-            requirements_path=prefixed_reqs_path,
+
+# TODO: Replace use of `repo` and `file_path` args with pcs.Path.Path since
+#     pcs.repo.Repo and pcs.path.RelativePath are now sub-types
+#     of pcs.Path.Path.
+class FileModule(Module):
+    def __init__(
+        self,
+        repo: Repo,
+        file_path: str,
+        imports: Dict[str, "Module"] = None,
+        virtual_env: VirtualEnv = None,
+    ):
+        """
+        :param repo: Repo where this Module's source file can be found. The
```

Comment:
High-level thought: it occurs to me that the `repo` terminology can be a bit misleading (esp. when the repo is just a local directory on the filesystem).  I wonder if there's a better word for this...

Reply:
Yeah. Maybe FileSource?